pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -441,10 +441,33 @@ func (m *kubeGenericRuntimeManager) makeMounts(opts *kubecontainer.RunContainerO
 			// open(2) to create the file, so the final mode used is "mode &
 			// ~umask". But we want to make sure the specified mode is used
 			// in the file no matter what the umask is.
-			if err := m.osInterface.Chmod(containerLogPath, 0666); err != nil {
+			if err := m.osInterface.Chmod(containerLogPath, 0660); err != nil {
 				utilruntime.HandleError(fmt.Errorf("unable to set termination-log file permissions %q: %v", containerLogPath, err))
 			}
 
+			var containerUid, containerGid int
+			if container.SecurityContext.RunAsUser != nil {
+				containerUid = int(*container.SecurityContext.RunAsUser)
+			} else if pod.Spec.SecurityContext.RunAsUser != nil {
+				containerUid = int(*pod.Spec.SecurityContext.RunAsUser)
+			} else {
+				containerUid = 0
```

Comment:
This probably doesn't work if the user is specified in the Dockerfile.

I _think_ there are other parts of Kubernetes (something to check if it doesn't run as root? Something else too, but I can't remember now) that contemplate all, not just this.

Maybe it is a reasonable trade-off, though, to only check that. I don't know

Reply:
> This probably doesn't work if the user is specified in the Dockerfile.

That is correct. I think it would be reasonable to expect end-user to configure SecurityContext explicitly in such case, but if there is a way to determine this automatically it would be better. I have not found anything that would allow to peak into the container image. The closest I was able to find was [`DetermineEffectiveRunAsUser(...)`](https://github.com/kubernetes/kubernetes/blob/d89d5ab2680bc74fe4487ad71e514f4e0812d9ce/pkg/securitycontext/util.go#L123), but it still only derives the information from the declared pod/container security context, not the image.